crates/sui-framework-build/src/compiled_package.rs

```diff
@@ -585,6 +585,19 @@ pub fn gather_dependencies(resolution_graph: &ResolvedGraph) -> PackageDependenc
     }
 }
 
+pub fn root_published_at(resolution_graph: &ResolvedGraph) -> Option<ObjectID> {
+    let root_package = resolution_graph
+        .package_table
+        .get(&resolution_graph.graph.root_package)
+        .unwrap();
+    root_package
+        .source_package
+        .package
+        .custom_properties
+        .get(&Symbol::from(PUBLISHED_AT_MANIFEST_FIELD))
+        .and_then(|v| ObjectID::from_str(v.as_str()).ok())
```

Comment:
Indeed we (should) have a strong guarantee there.  Technically we could do what you're saying, or even better, use the `?` syntax:

```rust
let published_at = resolution_graph
    .package_table
    .get(&resolution_graph.graph.root_package)?
    .source_package
    .package
    .custom_properties
    .get(&Symbol::from(PUBLISHED_AT_MANIFEST_FIELD))?;
ObjectID::from_str(v.as_str()).ok()
```

But I think it still makes sense to call `unwrap()` when getting the root_package, because it would be good to know if the thing we thought was true was not -- it suggests something quite badly wrong has happened.

Reply:
Yea, the root package should _always_ exist there. It's fundamental, and if it doesn't exist either (a) things should have already errored out before this; or (b) they're about to error out (or produce bad results at worst) because the package is seriously malformed and in an inconsistent state. 


I like the chaining (and question mark) suggestions! I'll incorporate before landing.